docs/book/src/tasks/experimental-features/cluster-class-operations.md

```diff
@@ -22,11 +22,11 @@ Once the changes are applied, the topology controller reacts as described in the
+| controlPlane.metadata                           | If labels/annotations are added, changed or deleted the ControlPlane objects are updated (in place update).<br /><br /> In case of KCP, corresponding controlPlane Machines are updated (rollout) only when adding or changing labels or annotations; deleted label should be removed manually from machines or they will go away automatically at the next machine rotation.      |
 | controlPlane.ref                                | Corresponding ControlPlane objects are updated (in place update). <br /> If updating ControlPlane objects implies changes in the spec, the corresponding ControlPlane Machines are updated accordingly (rollout).                                                                                                                                                                                                                                                                                          |
 | controlPlane.machineInfrastructure.ref          | If the referenced template has changes only in metadata labels or annotations, the corresponding InfrastructureMachineTemplates are updated (in place update). <br /> <br />If the referenced template has changes in the spec:<br />  - Corresponding InfrastructureMachineTemplate are rotated (create new, delete old)<br />  - Corresponding ControlPlane objects are updated with the reference to the newly created template (in place update)<br />  - The corresponding controlPlane Machines are updated accordingly (rollout). |
 | workers.machineDeployments                      | If a new MachineDeploymentClass is added, no changes are triggered to the Clusters. <br />If an existing MachineDeploymentClass is changed, effect depends on the type of change (see below).  <br /><br />Note: Deleting an existing MachineDeploymentClass is not supported.                                                                                                                                                                                                                                       |
-| workers.machineDeployments[].metadata           | If labels/annotations are added or their value is changed, MachineDeployment objects are updated (in place update) and corresponding worker Machines are updated (rollout). <br /> <br />If labels/annotations are deleted, the label is downgraded from managed to instance specific value, but no changes are applied. <br />In order to actually remove the label it is required for the user to change MachineDeployments object in place.                                                                         |
+| workers.machineDeployments[].metadata           | If labels/annotations are added, changed or deleted the MachineDeployment objects are updated (in place update) and corresponding worker Machines are updated (rollout).       |
```

Comment:
I wonder if we want to add something structured like https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/architecture/controllers/metadata-propagation.md#kubeadmcontrolplane so we can move them there once this is not experimental.

Reply:
TBH I think those are separated concern: this table is about what are the effects of changing a ClusterClass, and this impacts on all the Clusters using the Clusters Class. The page you are linking instead is about metadata propagation on CAPI objects inside a Cluster (and it applies both on managed and unmanaged objects).

Said that, nothing prevent us to duplicate information when ClusterClass graduates, but I don't think it makes sense to break down/alter the current table for the sake of copying a subset of info on another page sometime in the future.